torch/jit/_trace.py

```diff
@@ -971,17 +974,34 @@ def register_submods(mod, prefix):
                 func = getattr(mod, method_name)
                 argument_names = get_callable_argument_names(func)
 
-            example_inputs = make_tuple(example_inputs)
+            if isinstance(example_inputs, dict):
+                # Raise exception when the user provided key names are not aligned with forward() method's arguments' name/
+                for key in example_inputs:
+                    if key not in argument_names:
+                        valid_arguments = "[" + ','.join(argument_names) + "]"
```

Comment:
also commented above, but I think it would be better if this behavior was guarded by an additional parameter passed into torch.jit.trace.

Reply:
also, please add documentation of this flag to the documentation for trace_method.

And, if you want to expose this flag in torch.jit.trace(), I think it would be preferable to also support unpacking python dicts in torch.jit.trace for functions as well. But if you don't want to add that I think it's fine, in that case don't add the flag to torch.jit.trace.

Reply:
Yes, once we reach an agreement with the API design, I will do the corresponding modification and update the document.

Reply:
Yes, once we reach an agreement with the API design, I will do the corresponding modification and update the document.